### PR TITLE
feat: support x-html2md-img-src header

### PR DESCRIPTION
**Pull request by Buuhuu**
This header controls which image soruces can be trusted. html2md will fetch and upload all images from these allowed sources. Images from unallowed sources will be kept untouched.

Please ensure your pull request adheres to the following guidelines:
- [ ] make sure to link the related issues in this description
- [ ] when merging / squashing, make sure the fixed issue references are visible in the commits, for easy compilation of release notes

## Related Issues
- #268 

Thanks for contributing!

### CODE REVIEW

File: src/mdast-process-images.js [hunk 1]
```diff
@@ -12,28 +12,78 @@
...
+          return false;
+        }
+      };
+    }
+
+    return (urlStr) => {
+      try {
+        // omit the protocol check, as https:// is enforced earlier already
+        // omit the pathname check, as we may not need it (yet)
+        const url = new URL(urlStr);
```
Comment (tripodsan) [inline src/mdast-process-images.js:39]: same as as above: 

check for urls starting with `/`

```suggestion
        const url = new URL(urlStr, baseUrl);
```

File: src/mdast-process-images.js [hunk 2]
```diff
@@ -12,28 +12,78 @@
...
   // gather all image nodes
+  const filter = createFilter(log, baseUrl, imgSrcPolicy);
   const images = [];
   visit(tree, (node) => {
     if (node.type === 'image') {
       const { url = '' } = node;
-      if (url.startsWith('https://')) {
-        images.push(node);
-      } else if (url.indexOf(':') < 0) {
+      if (url.indexOf(':') < 0) {
```
Comment (tripodsan) [inline src/mdast-process-images.js:82]: I would always accept all relative urls...

File: src/index.js [hunk 3]
```diff
@@ -87,6 +87,7 @@ async function run(request, ctx) {
     }
   }
   const html = await res.text();
+  const imgSrcPolicy = (res.headers.get('x-html2md-img-src') || 'self').split(' ');
```
Comment (tripodsan) [inline src/index.js:90]: more tolerant split:

```suggestion
  const imgSrcPolicy = (res.headers.get('x-html2md-img-src') || 'self').split(/\s+/);
```

File: src/mdast-process-images.js (hunk 2 above)
Comment (Buuhuu) [reply src/mdast-process-images.js:82]: Which means we implicitly assume the 'self' img-src policy value and skip the unnecessary filtering in this case. done in https://github.com/adobe/helix-html2md/pull/272/commits/85533c454c1907ee7e7a563b1f31adfadf6e9df6

**Pull request closed without merging**